nameko/messaging.py

```diff
@@ -166,26 +182,15 @@ def retry_policy(self):
         return DEFAULT_RETRY_POLICY
 
     def setup(self):
-
-        exchange = self.exchange
-        queue = self.queue
+        if self.exchange is not None:
```

Comment:
It's fine to leave it like that. It was a suggestion in case it was possible and we could also set it up also in the initialiser method.